modules/widgets/googleplus-community-badge.php

```diff
@@ -0,0 +1,250 @@
+<?php
+
+/**
+ * Register the widget for use in Appearance -> Widgets
+ */
+add_action( 'widgets_init', 'jetpack_googleplus_community_badge_init' );
+
+function jetpack_googleplus_community_badge_init() {
+	register_widget( 'WPCOM_Widget_GooglePlus_Community_Badge' );
+}
+
+/**
+ * Google+ Community Badge widget class
+ * Display a Google+ Community Badge as a widget
+ * https://developers.google.com/+/web/badge/
+ */
+class WPCOM_Widget_GooglePlus_Community_Badge extends WP_Widget {
+
+	private $default_width       = 300;
+	private $max_width           = 450;
+	private $min_width_portrait  = 180;
+	private $min_width_landscape = 273;
+	private $min_width;
+	private $default_theme       = 'light';
+	private $allowed_themes      = array('light', 'dark');
+	private $default_layout      = 'portrait';
+	private $allowed_layouts     = array('landscape', 'portrait');
+
+	function __construct() {
+		$this->min_width = min( $this->min_width_portrait, $this->min_width_landscape );
+
+		parent::__construct(
+			'googleplus-community-badge',
+			apply_filters( 'jetpack_widget_name', __( 'Google+ Community Badge', 'jetpack' ) ),
+			array(
+				'classname' => 'widget_googleplus_community_badge',
+				'description' => __( 'Display a Google+ Community Badge to connect visitors to your Google+ community', 'jetpack' )
+			)
+		);
+	}
+
+	function widget( $args, $instance ) {
+
+		extract( $args );
+
+		$like_args = $this->normalize_googleplus_args( $instance['like_args'] );
+
+		if ( empty( $like_args['href'] ) || ! $this->is_valid_googleplus_url( $like_args['href'] ) ) {
+			if ( current_user_can('edit_theme_options') ) {
+				echo $before_widget;
+				echo '<p>' . sprintf( __( 'It looks like your Google+ URL is incorrectly configured. Please check it in your <a href="%s">widget settings</a>.', 'jetpack' ), admin_url( 'widgets.php' ) ) . '</p>';
+				echo $after_widget;
+			}
+			echo '<!-- Invalid Google+ Community URL -->';
+			return;
+		}
+
+
+		$title    = apply_filters( 'widget_title', $instance['title'] );
+
+		$like_args['show_photo']   = (bool) $like_args['show_photo']   ? 'true' : 'false';
+		$like_args['show_owners']  = (bool) $like_args['show_owners']  ? 'true' : 'false';
+		$like_args['show_tagline'] = (bool) $like_args['show_tagline'] ? 'true' : 'false';
+
+		echo $before_widget;
+
+		if ( ! empty( $title ) ):
+			echo $before_title;
+
+			$badge_widget_title = '<a href="' . esc_url( $like_args['href'] ) . '">' . esc_html( $title ) . '</a>';
+
+			echo apply_filters( 'jetpack_googleplus_community_badge_title', $badge_widget_title, $title, $like_args['href'] );
```

Comment:
Could you add some documentation before the filter, as explained [here](https://make.wordpress.org/core/handbook/inline-documentation-standards/php-documentation-standards/#4-hooks-actions-and-filters)?